scipy/signal/signaltools.py

```diff
@@ -1020,8 +1020,9 @@ def convolve2d(in1, in2, mode='full', boundary='fill', fillvalue=0):
     val = _valfrommode(mode)
     bval = _bvalfromboundary(boundary)
 
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore', np.ComplexWarning)
+    with suppress_warnings() as sup:
```

Comment:
Should be easier to do elsewhere anyway, but why is this not done, it feels like there is probably some weird reason, but can't see it :):
```diff
diff --git a/scipy/signal/sigtoolsmodule.c b/scipy/signal/sigtoolsmodule.c
index 51842c806..543507780 100644
--- a/scipy/signal/sigtoolsmodule.c
+++ b/scipy/signal/sigtoolsmodule.c
@@ -1049,9 +1049,7 @@ static PyObject *sigtools_convolve2d(PyObject *NPY_UNUSED(dummy), PyObject *args
                                                                  zeros);
        }
        else {
-           afill = (PyArrayObject *)PyArray_FromObject(fill_value, NPY_CDOUBLE, 0, 0);
-           if (afill == NULL) goto fail;
-           newfill = (PyArrayObject *)PyArray_Cast(afill, typenum);
+           newfill = (PyArrayObject *)PyArray_FromObject(fill_value, typenum, 0, 0);
        }
        if (newfill == NULL) goto fail;
     }
```